### PR TITLE
[ENDPOINT] GET / news / OT102-45

### PR DESCRIPTION
**Pull request by gabyMangoni**
OT102-11
--------------------------------------------------------------------------------------------------------------------------------------------
**Description**
HOW: admin user
I WANT: to list the News
TO: be able to manage the News

--------------------------------------------------------------------------------------------------------------------------------------------

**Criteria of acceptance:**
GET a / news. It will return the list of entries whose type field is "news". It will return only the name, image and createdAt fields

--------------------------------------------------------------------------------------------------------------------------------------------

**Evidence**

list-news

### CODE REVIEW

File: controllers/entry.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  get: catchAsync(async (req, res, next) => {
+    try {
+      const allEntries = await getEntries()
+      endpointResponse({
+        res,
+        msg: 'Entries were retrieved successfully.',
+        body: allEntries,
+      })
+    } catch (error) {
+      next(new ErrorObject(`[Error retrieving contacts] - [contacts - get]: ${error.message}`, 404))
```
Comment (dvelezroman) [inline controllers/entry.js]: ` next(new ErrorObject(`[Error retrieving entries] - [entries - get]: ${error.message}`, 404))`

File: services/entry.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const { Entry } = require('../models')
+
+exports.getEntries = async () => {
+  try {
+    const entries = await Entry.findAll({
+      attributes: {
+        exclude: ['id', 'content', 'categoryId', 'type', 'updatedAt', 'deletedAt'],
```
Comment (dvelezroman) [inline services/entry.js]: I think... we need id, content, categoryId, type included in results....

**Review by dvelezroman — CHANGES_REQUESTED** [review]
Nice PR.. well formated.. just little changes requested

File: routes/entries.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const express = require('express')
+const { get } = require('../controllers/entry')
+
+const router = new express.Router()
+
+/* GET users listing. */
```
Comment (JonatanGarbuyo) [inline routes/entries.js]: /* GET entries listing. */

ojo el copy&paste  ;)

File: controllers/entry.js (hunk 1 above)
Comment (gabyMangoni) [reply controllers/entry.js]: Ready!

File: services/entry.js (hunk 2 above)
Comment (gabyMangoni) [reply services/entry.js]: Ready!

**Review by dvelezroman — APPROVED** [review]
Excellent

**Review by gabodmc — APPROVED** [review]
Ok!

File: services/entry.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const { Entry } = require('../models')
+
+exports.getEntries = async () => {
+  try {
+    const entries = await Entry.findAll({
+      attributes: {
+        exclude: ['updatedAt', 'deletedAt'],
+      },
```
Comment (joselo85) [inline services/entry.js:9]: Gabi, as was stated on the PR descritpion, you should only get entries that are only of the "news" type. On the same table we will have other types of entries such as: "testimonials".

**Review by alexiscjscab — APPROVED** [review]

**Review by joselo85 — APPROVED** [review]

**Review by JonatanGarbuyo — APPROVED** [review]

**Pull request merged**